Make `JoinKeys` related changes due to refactor (#486)
insightsengineering/teal.slice: NEWS.md

* Specified minimal version of package dependencies.
* Removed storing and restoring of `teal_slices` objects.

> * Update documentation and code to reflect `teal.data::JoinKeys` class and logic change to `teal.data::join_keys`.

Reviewer: do you mean refactor or remove?
```suggestion
* Update documentation and code to refactor/remove `teal.data::JoinKeys` class and logic change to `teal.data::join_keys`.
```

Author: How about this?

```suggestion
* Update documentation and code to reflect the changes due to the refactor of `teal.data::JoinKeys` into `teal.data::join_keys`.
```

Reviewer: its  better
